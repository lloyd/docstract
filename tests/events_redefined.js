// docstract supports a somewhat generic notion of "events".  These may
// be associated with a class or the top level file.  An event has a
// name, description, and payload.  That's it.

/**
 * @event progress
 * Allows the listener to understand approximately how much of the
 * page has loaded.
 * @payload {number} The percentage (0..100) of page load that is complete
 */
this.eventEmitter._emit("progress", curProgress);

/**
 * @event progress
 * Allows the listener to understand approximately how much of the
 * page has loaded.
 * @payload {number} The percentage (0..100) of page load that is complete
 */
this.eventEmitter._emit("progress", curProgress);
