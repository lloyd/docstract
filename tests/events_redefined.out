{
  "args": [
    "'progress' event redefined at line 13"
  ], 
  "exception_type": "<type 'exceptions.RuntimeError'>", 
  "stack": [
    [
      "<module>", 
      "got = ds.extractFromFile(os.path.join(testDir, test + \".js\"))"
    ], 
    [
      "extractFromFile", 
      "data = self.extract(contents)"
    ], 
    [
      "extract", 
      "self._analyzeBlock(block, context, firstBlock, stack, lineStart, line)"
    ], 
    [
      "_analyzeBlock", 
      "parseData['blockHandler'].merge(doc, stack[-1][1], guessedName, self._whatContext(stack))"
    ], 
    [
      "merge", 
      "raise RuntimeError(\"'%s' event redefined\" % doc[\"name\"])"
    ]
  ]
}
